extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/watermark.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Watermark component
+ *
+ * @module Watermark
+ */
+var React = require('react'),
+    CONSTANTS = require('../constants/constants'),
+    Utils = require('../components/utils'),
+    ClassNames = require('classnames');
+
+var Watermark = React.createClass({
+
+  handleWatermarkClick: function() {
+    if (this.props.playerState == CONSTANTS.STATE.PLAYING){
+        this.props.controller.togglePlayPause();
+    }
+  },
+
+  render: function() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 15 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)